Stored objects update instructions (#580)
cert-manager/website: content/en/docs/release-notes/release-notes-1.4.md

/ 3; else rb = renewBefore`. This change fixes a bug where, if a certificate's
duration is very slightly larger than `renewBefore` period, then a cert gets
renewed at `notAfter - renewBefore` which can lead to a continuous renewal loop,
see [`cert-manager#3897`](https://github.com/jetstack/cert-manager/issues/3897).

## Updating cert-manager CRDs and stored versions of cert-manager custom resources

> We have deprecated `cert-manager.io/v1alpha2`, `cert-manager.io/v1alpha3`, `cert-manager.io/v1beta1`, `acme.cert-manager.io/v1alpha2`, `acme.cert-manager.io/v1alpha3`, `acme.cert-manager.io/v1beta1` APIs.

Reviewer: suggestion: would this be a little more readable as a list?

e.g.

```suggestion
We've deprecated the following APIs: 

- `cert-manager.io/v1alpha2`
- `cert-manager.io/v1alpha3`
- `cert-manager.io/v1beta1`
- `acme.cert-manager.io/v1alpha2`
- `acme.cert-manager.io/v1alpha3`
- `acme.cert-manager.io/v1beta1`

```

These APIs will be removed in `cert-manager` `v1.6`.

> If you have a `cert-manager` installation that is using or has previously used these APIs you may need to update `cert-manager` custom resources and CRDs. This needs to be done before upgrading to `cert-manager` `v1.6`. (See Kubernetes docs for more detailed explanation [here](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version))

Reviewer: suggestion: minor wording change
nitpick: we refer to "`cert-manager`" here, but we frequently use "cert-manager" in the documentation, e.g. in https://cert-manager.io/docs/release-notes/release-notes-1.3/ - should we follow the existing style? (this applies elsewhere in the commit)

```suggestion
If you have a cert-manager installation that is using or has previously used these deprecated APIs you may need to update your cert-manager resources and CRDs. This needs to be done before upgrading to cert-manager `v1.6`. (See the [Kubernetes docs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version) for a more detailed explanation)
```

Steps:

> 1. Make sure your `cert-manager` deployment is `v1` or later.

Reviewer: suggestion: this is referring to the installed version of cert-manager, right? if so, would it be good to give the exact version number as it appears to people, to make it exactly clear which version they need?

```suggestion
1. Make sure your installed `cert-manager` deployment is `v1.0.0` or later.
```

> 2. Make sure any version-controlled `cert-manager` custom resource config files updated to use `cert-manager.io/v1` API and re-applied. This should update stored versions for the given resources.

Reviewer: nitpick: missing word

```suggestion
2. Make sure any version-controlled `cert-manager` custom resource config files are updated to use the `cert-manager.io/v1` API and re-applied. This should update stored versions for the given resources.
```

> Further please follow the official Kubernetes documentation [here](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version):
> 
> 1. If there are other resources (`ingress-shim` `Certificate`s, `CertificateRequest`s etc that aren't stored in Git and may have been created using the deprecated APIs, you can run  `kubectl get <resource_name> -oyaml | kubectl apply -f -` to force the resources to be stored in `etcd` at the current storage version.
> 2. To remove legacy API versions from `cert-manager` CRDs, you can run the following `curl` commands:

Reviewer: suggestion: a slight rewording; I used `\` only to escape the code block.

(I also wouldn't personally expect the documentation to use "please" here, but that's a nitpick)

```suggestion
After that, follow the official Kubernetes documentation [here](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version):

1. If there are other resources (e.g. `Certificate`s created by ingress-shim, `CertificateRequest`s, etc.) that aren't stored in Git and may have been created using the deprecated APIs, run
   \```bash
   kubectl get <resource_name> -oyaml | kubectl apply -f -
   \```
   to force the resources to be stored in `etcd` at the current storage version.

2. To remove legacy API versions from `cert-manager` CRDs run the following `curl` commands:
```

> ```

Reviewer: nitpick: I think it's always best to specify a type for code blocks, e.g.

```suggestion
```bash
```
